pkg/controller/deactivation/deactivation_controller.go

```diff
@@ -149,6 +165,30 @@ func (r *ReconcileDeactivation) Reconcile(request reconcile.Request) (reconcile.
 	logger.Info("user account time values", "deactivation timeout duration", deactivationTimeout, "provisionedTimestamp", provisionedTimestamp)
 
 	timeSinceProvisioned := time.Since(provisionedTimestamp.Time)
+
+	deactivatingNotificationTimeout := time.Duration((deactivationTimeoutDays - r.config.GetUserSignupDeactivatingNotificationDays()) * 24)
+
+	if timeSinceProvisioned < deactivatingNotificationTimeout {
+		// It is not yet time to send the deactivating notification so requeue until it will be time to send it
+		requeueAfterTimeToNotify := deactivatingNotificationTimeout - timeSinceProvisioned
+		logger.Info("requeueing request", "RequeueAfter", requeueAfterTimeToNotify, "Expected deactivating notification date/time", time.Now().Add(requeueAfterTimeToNotify).String())
+		return reconcile.Result{RequeueAfter: requeueAfterTimeToNotify}, nil
+	}
+
+	if condition.IsNotTrue(usersignup.Status.Conditions, toolchainv1alpha1.UserSignupUserDeactivatingNotificationCreated) {
+		if err := r.sendDeactivatingNotification(logger, usersignup); err != nil {
+			logger.Error(err, "Failed to create user deactivating notification")
+
+			// set the failed to create notification status condition
+			return reconcile.Result{}, r.WrapErrorWithStatusUpdate(logger, usersignup, r.SetStatusDeactivatingNotificationCreationFailed, err, "Failed to create user deactivating notification")
+		}
+
+		if err := r.UpdateStatus(logger, usersignup, r.SetStatusDeactivatingNotificationCreated); err != nil {
+			logger.Error(err, "Failed to update notification created status")
+			return reconcile.Result{}, err
+		}
+	}
+
 	if timeSinceProvisioned < deactivationTimeout {
```

Comment:
As I already mentioned in the slack channel, we should count 3 days from the time when the deactivating notification was sent and then deactivate.

Reply:
Yeah it would be a better UX if we deactivate in 3 days after the actual email/notification. So imagine we have an account which is to be deactivated in 4 days. And then we have an outage for 4 days. When the service is up again we would like to avoid the situation when we send a notification saying the account will be deactivated in 3 days and then immediately deactivate it. It would be OK to postpone the actual deactivation even if it's past due.
But I'm a bit concerned about suck accounts if for some reason we failed to create a notification resource... Not sure if we should really be concerned about that since we would have a bigger problem if we can't/don't create notification resources for extended period of time.

Reply:
Agreed we should start the deactivation timer from when the pre notification is sent out.  I'm currently working on the changes for this now, it's going to break a lot of our tests.